src/nat/agent/agentspec/config.py

```
@@ -0,0 +1,64 @@
+
+    tool_names: list[FunctionRef | FunctionGroupRef] = Field(
+        default_factory=list, description="Optional list of tool names/groups to expose to the Agent Spec runtime.")
+
+    max_history: int = Field(default=15, description="Maximum number of messages to keep in conversation history.")
+
+    @model_validator(mode="after")
+    def _validate_sources(self):
+        provided = [self.agentspec_yaml, self.agentspec_json, self.agentspec_path]
+        cnt = sum(1 for v in provided if v)
+        if cnt != 1:
+            raise ValueError("Exactly one of agentspec_yaml, agentspec_json, or agentspec_path must be provided")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)